woocommerce/woocommerce-ios
# Shipping Labels: Update purchase form view model to support multi-package

Part of #4599

# Description
To prepare for updates of the purchase form fields to support multi-package, I think it's best to first update the form view model. This PR aims to replace the single selected package ID and weight with a list of basic package details. This also leads to updates in the view controller and package details.

# Changes
## ShippingLabelFormViewModel
- Replaced `selectedPackageID` and `totalPackageWeight` with `selectedPackagesDetails`. This is a list with of a new model `ShippingLabelPackageInfo` with basic details: package ID, total weight and product ID list. Note: I'm not quite happy with the naming of the model and the variable so any suggestion for renaming is appreciated!
- Replaced `selectedPackage` with `selectedPackages` array and updated its calculated value to prepare appropriate model for each package ID.
- Updated other parts relating to selected packages: handling package selection, setting up default customs forms, purchasing label.

## ShippingLabelFormViewController & ShippingLabelPackageDetails
- Updated input and output of `ShippingLabelPackageDetails` to use the selected package list details.

## ShippingLabelPackageDetailsViewModel
- Updated initializer with new parameter `selectedPackages`. 
- Added temporary updates to support the new parameter. These should be fixed in the next PR to properly support multiple packages.

# Testing
With the changes in place, it is expected that configuring package details should still work properly as before. Fetching carriers and purchasing label should also work as before.

1. Make sure that your test store has installed WCShip plugin and configured packages including DHL packages in WPAdmin > WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Open an order detail that is eligible for the shipping label.
3. Configure Ship To and Ship From.
4. Configure Package Details: select any package and / or update package weight manually and tap Done. Notice that the details are displayed correctly on the purchase form.
5. Tap Package Details again, notice that correct package and weight are displayed.
6. Tap Carrier and Rates, notice that carrier list can still be fetched for the package if the details are appropriate.
7. Proceed through Payment Method and purchase the label. The purchase should still succeed.

Update release notes:
